packages/site/src/hooks/usePageValues.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMemo, useState } from "react";
+import { InputText, Switch, Text } from "@jobber/components";
+import { ValueStateInternals } from "../types/services";
+import { ContentExport } from "../types/content";
+import { SelectWithOptions } from "../components/SelectWithOptions";
+
+/**
+ * This is doing a few different things.
+ *
+ * 1. Any change to the PropList will call updateValue, which will update the values state.
+ * 2. The stateValues are memoized to prevent re-renders and modified to the DataList format.
+ * 3. Any value that is a straight function is converted to a real function so it can be used in the actual component.
+ * 4. There are two utility functions that may move out of here:
+ *   - isFunction: Checks if a string is a function
+ *  - determineListComponent: Determines what component to use based on the type of the prop
+ * @param meta Content Meta
+ * @returns Values to be used on the ContentView page
+ */
+export const usePageValues = (meta: ContentExport) => {
+  const [values, setValues] = useState(meta.component.defaultProps);
+
+  const updateValue = (key: string, value: string | number | boolean) => {
+    if (key) {
+      setValues(prev => ({
+        ...prev,
+        [key]: value as string,
+      }));
+    }
+  };
+
+  const stateValues = useMemo(() => {
+    return meta.props.map(propList => {
+      return {
+        name: propList.displayName,
+        props: Object.keys(propList.props).map((key, index) => {
+          const prop = propList.props[key];
+
+          return {
+            id: index,
+            key,
+            required: prop?.required ? "*" : "",
+            description: prop?.description,
+            component: determineListComponent({
+              key,
+              type: prop?.type.name || "",
+              placeholder: key,
+              value: values[key] as string,
+              updateValue,
+            }),
+          };
+        }),
+      };
+    });
+  }, [values]);
+
+  const valuesWithFunctions: Record<
```

Comment:
I should have more comments here. This loops through the provided values, and sees if any of them fit the shape of a function.

If they do, we transform the function so it actually executes when the associated callback fires on the component.